ipaserver/install/httpinstance.py

```diff
@@ -436,41 +321,50 @@ def __setup_ssl(self):
                 certmonger.request_and_wait_for_cert(
-                    certpath=db.secdir,
-                    nickname=self.cert_nickname,
+                    certpath=(paths.HTTPD_CERT_FILE, paths.HTTPD_KEY_FILE),
                     principal=self.principal,
-                    passwd_fname=db.passwd_fname,
                     subject=str(DN(('CN', self.fqdn), self.subject_base)),
                     ca='IPA',
                     profile=dogtag.DEFAULT_PROFILE,
                     dns=[self.fqdn],
-                    post_command='restart_httpd')
+                    post_command='restart_httpd',
+                    storage='FILE',
+                )
             finally:
                 if prev_helper is not None:
                     certmonger.modify_ca_helper('IPA', prev_helper)
-
-            self.cert = db.get_cert_from_db(self.cert_nickname)
+            self.cert = x509.load_certificate_from_file(
+                paths.HTTPD_CERT_FILE
+            )
 
             if prev_helper is not None:
                 self.add_cert_to_service()
 
             # Verify we have a valid server cert
-            server_certs = db.find_server_certs()
-            if not server_certs:
-                raise RuntimeError("Could not find a suitable server cert.")
+            # FIXME: come up with openssl equivalent
+            #  server_certs = db.find_server_certs()
+            #  if not server_certs:
+            #      raise RuntimeError("Could not find a suitable server cert.")
 
         # store the CA cert nickname so that we can publish it later on
-        self.cacert_nickname = db.cacert_name
-
-    def __import_ca_certs(self):
-        db = certs.CertDB(self.realm, nssdir=paths.HTTPD_ALIAS_DIR,
-                          subject_base=self.subject_base)
-        self.import_ca_certs(db, self.ca_is_configured)
+        # self.cacert_nickname = db.cacert_name
+        # FIXME: figure this out too
+        sysupgrade.set_upgrade_state('ssl.conf', 'migrated_to_mod_ssl', True)
+
+    def configure_mod_ssl_certs(self):
+        """Configure the mod_ssl certificate directives"""
+        installutils.set_directive(paths.HTTPD_SSL_CONF,
+                                   'SSLCertificateFile',
+                                   paths.HTTPD_CERT_FILE, False)
+        installutils.set_directive(paths.HTTPD_SSL_CONF,
+                                   'SSLCertificateKeyFile',
+                                   paths.HTTPD_KEY_FILE, False)
+        installutils.set_directive(paths.HTTPD_SSL_CONF,
+                                   'SSLCACertificateFile',
+                                   paths.IPA_CA_CRT, False)
 
     def __publish_ca_cert(self):
-        ca_db = certs.CertDB(self.realm, nssdir=paths.HTTPD_ALIAS_DIR,
-                             subject_base=self.subject_base)
-        ca_db.export_pem_cert(self.cacert_nickname, paths.CA_CRT)
+        self.import_ca_certs_file(paths.CA_CRT, self.ca_is_configured)
```

Comment:
Previously, this code would only publish the last CA cert in the chain, now it pastes the whole chain there. This is most probably used only in legacy clients and as a very improbable fallback to receiving the CA cert in `ipa-client-install`.
I wonder - could this behavior change have negative effect in such scenarios? Or should we have actually had the full CA chain there from the beginning?

Reply:
See https://pagure.io/freeipa/issue/6454

A CA trust store should only contain trust anchors, not intermediate CA certs. Since most of our infrastructure doesn't support AIA chasing and AIA chasing has also a negative performance impact, intermediate CA certs must be part of the the cert chain in TLS handshake.